src/Features/LanguageServer/Protocol/Handler/SemanticTokens/SemanticTokensHelpers.cs

```diff
@@ -0,0 +1,243 @@
+                [ClassificationTypeNames.FieldName] = LSP.SemanticTokenTypes.Property, // TO-DO: Potentially change to custom type
+                [ClassificationTypeNames.Identifier] = LSP.SemanticTokenTypes.Variable, // TO-DO: Potentially change to custom type
+                [ClassificationTypeNames.InterfaceName] = LSP.SemanticTokenTypes.Interface,
+                [ClassificationTypeNames.Keyword] = LSP.SemanticTokenTypes.Keyword,
+                [ClassificationTypeNames.LabelName] = LSP.SemanticTokenTypes.Variable, // TO-DO: Potentially change to custom type
+                [ClassificationTypeNames.LocalName] = LSP.SemanticTokenTypes.Member, // TO-DO: Potentially change to custom type
+                [ClassificationTypeNames.MethodName] = LSP.SemanticTokenTypes.Member, // TO-DO: Potentially change to custom type
+                [ClassificationTypeNames.ModuleName] = LSP.SemanticTokenTypes.Member, // TO-DO: Potentially change to custom type
+                [ClassificationTypeNames.NamespaceName] = LSP.SemanticTokenTypes.Namespace,
+                [ClassificationTypeNames.NumericLiteral] = LSP.SemanticTokenTypes.Number,
+                [ClassificationTypeNames.Operator] = LSP.SemanticTokenTypes.Operator,
+                [ClassificationTypeNames.OperatorOverloaded] = LSP.SemanticTokenTypes.Operator, // TO-DO: Potentially change to custom type
+                [ClassificationTypeNames.ParameterName] = LSP.SemanticTokenTypes.Parameter,
+                [ClassificationTypeNames.PreprocessorKeyword] = LSP.SemanticTokenTypes.Keyword, // TO-DO: Potentially change to custom type
+                [ClassificationTypeNames.PreprocessorText] = LSP.SemanticTokenTypes.String, // TO-DO: Potentially change to custom type
+                [ClassificationTypeNames.PropertyName] = LSP.SemanticTokenTypes.Property,
+                [ClassificationTypeNames.Punctuation] = LSP.SemanticTokenTypes.Operator, // TO-DO: Potentially change to custom type
+                [ClassificationTypeNames.RegexAlternation] = LSP.SemanticTokenTypes.Regexp, // TO-DO: Potentially change to custom type
+                [ClassificationTypeNames.RegexAnchor] = LSP.SemanticTokenTypes.Regexp, // TO-DO: Potentially change to custom type
+                [ClassificationTypeNames.RegexCharacterClass] = LSP.SemanticTokenTypes.Regexp, // TO-DO: Potentially change to custom type
+                [ClassificationTypeNames.RegexComment] = LSP.SemanticTokenTypes.Regexp, // TO-DO: Potentially change to custom type
+                [ClassificationTypeNames.RegexGrouping] = LSP.SemanticTokenTypes.Regexp, // TO-DO: Potentially change to custom type
+                [ClassificationTypeNames.RegexOtherEscape] = LSP.SemanticTokenTypes.Regexp, // TO-DO: Potentially change to custom type
+                [ClassificationTypeNames.RegexQuantifier] = LSP.SemanticTokenTypes.Regexp, // TO-DO: Potentially change to custom type
```

Comment:
this surprises me.  In typescript there is some amount of regexp classification:

![image](https://user-images.githubusercontent.com/4564579/90663870-0f6e0500-e1ff-11ea-8cee-34475b74541c.png)

So it feels like LSP must have some way of distinguishing at least some regexp info... can you look into that later?

Reply:
That's super interesting, I'll definitely look into this as a follow-up after this PR is merged. David mentioned that Razor is hardcoding mappings... not sure if TS is doing something similar?

Reply:
Without custom tokens there are no other Regex types. Typescript in VSCode doesn't emit semantic tokens for regex, so the editor shows classifications from the textmate language.